qadence/operations.py

```diff
@@ -169,13 +170,35 @@ def dagger(self) -> Z:
         return self
 
 
-class N(PrimitiveBlock):
+class Projector(ProjectorBlock):
+    """The projector operator."""
+
+    name = OpName.PROJ
+
+    def __init__(
+        self,
+        ket: str,
+        bra: str,
+        qubit_support: int | tuple[int, ...],
+    ):
+        super().__init__(ket=ket, bra=bra, qubit_support=qubit_support)
+
+    @property
+    def generator(self) -> None:
+        raise ValueError("Property `generator` not available for non-unitary operator.")
+
+    @property
+    def eigenvalues_generator(self) -> None:
+        raise ValueError("Property `eigenvalues_generator` not available for non-unitary operator.")
+
+
+class N(Projector):
     """The N = (1/2)(I-Z) operator."""
 
     name = OpName.N
 
-    def __init__(self, target: int):
-        super().__init__((target,))
+    def __init__(self, target: int, state: str = "1"):
+        super().__init__(ket=state, bra=state, qubit_support=(target,))
```

Comment:
Wasn't more efficient to keep the previous implementation and leave the `ProjectorBlock` just for generic projectors?

Reply:
Well, it felt natural to express all possible operators using the projectors whenever possible ? I was thinking along Vincent's comment about helper functions for the number operator. I also tried to use `N` everywhere possible which makes for a clean interface I think ?

Reply:
Yeah, actually thinking about it again, it makes sense. As long as it does not break something else. However, now it is not anymore the `N` operator, technically.